### PR TITLE
Add sphinx docs

### LINKED ISSUE #12: Add Docs
Add Sphinx documentation that we can use with http://readthedocs.org

### PR DESCRIPTION
**Pull request by damonkelley**
Will resolve #12

### CODE REVIEW

**Comment by damonkelley** [comment]
The current draft of this PR can be viewed here

http://django-name.readthedocs.org/en/add-sphinx-docs/

**Comment by damonkelley** [comment]
Add an env to the build for docs to be built

**Comment by damonkelley** [comment]
ping @somexpert @ldko 

This is a sort of initial draft of the docs. Once we get this merged in, we can keep it up to date as more features and configurations options are made available.

File: docs/development.rst [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    $ docker-compose run --rm web tox
+
+The Tox configuration will test this app with Django 1.6 - 1.8.
+
+To run the tests only with development environment (i.e. with Django 1.7) ::
+
+    $ docker-compose run --rm web ./runtests.py
+
+.. note::
+    This is the same command that Tox issues inside each test environment is has defined.
```
Comment (ldko) [inline docs/development.rst]: "is has defined" should be changed here and in README

File: docs/index.rst [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+   You can adapt this file completely to your liking, but it should at least
+   contain the root `toctree` directive.
+
+Introduction
+============
+
+About
+-----
+
+The Name App is a tool for originally developed for documenting names by the UNT Libraries for use in its various digital library systems and collections. The app provides a consistent way of communicating the authorized version of a name and information about the name that is useful for reuse. The Name App generates a unique URL for each name that can be used to unambiguously refer to a person, organization, event or piece of software. In addition to an HTML page for each name there are a number of other formats available for each record including a MADS XML version and a simple JSON representation. A key feature of the Name App is the ability to link to other vocabularies such as the Virtual International Authority File (VIAF), the Library of Congress Name Authority File, or Wikipedia.
```
Comment (ldko) [inline docs/index.rst]: Change "The Name App is a tool for originally developed for"

Also, "refer to a person, organization, event or piece of software" doesn't mention building.

File: docs/models.rst [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+======
+
+- :ref:`name-model-ref`
+- :ref:`identifier-type-model-ref`
+
+.. _name-model-ref:
+
+Name
+----
+Name objects have a variety configurable options.
```
Comment (ldko) [inline docs/models.rst]: Change to "a variety of configurable options"?

File: docs/models.rst [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+Name objects have a variety configurable options. 
+
+Fields
+''''''
+
+``name`` - The canonical form of the name.
+
+``name_type`` - One of `Personal`, `Organization`, `Software`, `Building`, or `Event`
+
+``biography`` - Markdown enabled biography of the personal the Name record represents.
```
Comment (ldko) [inline docs/models.rst]: I think "of the personal" is supposed to be something else.

File: docs/models.rst [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+
+``begin`` - The starting date for the Name record. This will be different for each Name Type, for instance, this field would be the birth date for a `Personal` name, and the erected date for a `Building` name.
```
Comment (ldko) [inline docs/models.rst]: I don't think there should be a comma in here: "for a `Personal` name, and the erected date"

File: docs/models.rst [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+
+- `Acronym` 
+- `Abbreviation`
+- `Expansion`
+- `Translation`
+- `Other`
+
+Identifiers
+'''''''''''
+The Identifier contains an type, (see :ref:`identifier-type-model-ref`), and value which is often represented as a permalink. For instance, the link to the person's Twitter profile would be an Identifier.
```
Comment (ldko) [inline docs/models.rst]: not "an type"

File: docs/models.rst [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+
+- `Biographical/Historical`
+- `Deletion Information`
+- `Nonpublic`
+- `Source`
+- `Other`
+
+Locations
+'''''''''
+Locations are represented by a geographic coordinate, wich enable some mapping features within the app when present. A Name's location may be either ``current`` or ``former``, and a Name may only have one ``current`` location at any given time.
```
Comment (ldko) [inline docs/models.rst]: "wich"

File: docs/models.rst [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+
+Misc Options
+''''''''''''
+
+Name records are capable of being merged with other Name records. Once merged with another record, any attempts to retrieve information about the merged record with redirect users to the Name record the was the target of the merge.
```
Comment (ldko) [inline docs/models.rst]: "with redirect users" should be "will"?

File: docs/models.rst (hunk 4 above)
Comment (damonkelley) [inline docs/models.rst]: How about 

> of the entity for which the Name record represents.

?

Comment (ldko) [inline docs/models.rst]: or "of the entity that the Name record represents"

**Comment by damonkelley** [comment]
@ldko, those changes have been completed.

File: docs/index.rst [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+   You can adapt this file completely to your liking, but it should at least
+   contain the root `toctree` directive.
+
+Introduction
+============
+
+About
+-----
+
+The Name App is a tool originally developed for documenting names by the UNT Libraries for use in its various digital library systems and collections. The app provides a consistent way of communicating the authorized version of a name and information about the name that is useful for reuse. The Name App generates a unique URL for each name that can be used to unambiguously refer to a person, organization, event building or piece of software. In addition to an HTML page for each name there are a number of other formats available for each record including a MADS XML version and a simple JSON representation. A key feature of the Name App is the ability to link to other vocabularies such as the Virtual International Authority File (VIAF), the Library of Congress Name Authority File, or Wikipedia.
```
Comment (ldko) [inline docs/index.rst]: "event building"

File: README.md [hunk 10]
```diff
@@ -3,9 +3,9 @@
...
+The Name App is a tool originally developed for documenting names used by the UNT Libraries in its various digital library systems and collections. The app provides a consistent way of communicating the authorized version of a name and information about the name that is useful for reuse. The Name App generates a unique URL for each name that can be used to unambiguously refer to a person, organization, event or piece of software. In addition to an HTML page for each name there are a number of other formats available for each record including a MADS XML version and a simple JSON representation. A key feature of the UNT Name App is the ability to link to other vocabularies such as the Virtual International Authority File (VIAF), the Library of Congress Name Authority File, or Wikipedia.
```
Comment (ldko) [inline README.md]: add "building" for what can be referred to by a name

**Comment by ldko** [comment]
:+1:

**Pull request merged**